Core/Core/Helpers/Http.cs

```diff
@@ -52,23 +52,26 @@ public static AsyncRetryPolicy<HttpResponseMessage> HttpAsyncPolicy(IEnumerable<
   /// <returns>True if the user is connected to the internet, false otherwise.</returns>
   public static async Task<bool> UserHasInternet()
   {
-    //can ping cloudfare, skip further checks
-    //this method should be the fastest
-    if (await Ping("1.1.1.1").ConfigureAwait(false))
+    string? defaultServer = null;
+    try
     {
+      //Perform a quick ping test e.g. to cloudflaire dns, as is quicker than pinging server
+      if (await Ping("1.1.1.1").ConfigureAwait(false))
+      {
+        return true;
+      }
+
+      defaultServer = AccountManager.GetDefaultServerUrl();
+      Uri serverUrl = new(defaultServer);
+      await HttpPing(serverUrl).ConfigureAwait(false);
       return true;
```

Comment:
Changes are good, but this function is doing more than it should (possibly another ticket?).
`UserHasInternet` as it stands requires your default account server to be online too, which is not what the function says it will do. Checking if the "computer user" has internet is not the same as checking if the "speckle user" is valid against the server, specially since the default server url can be other than our own.

This will result in "You are not connected to the internet" problems that may not be caused by that.

Reply:
100%, I was holding back on re-writing this code. We're relying on either `1.1.1.1` ping succeeding (sometimes blocked in corporate firewall/DNS) or your default server being online and your account being valid for it.

Reply:
Resolving, added ticket to re-evaluate this in the backlog
https://spockle.atlassian.net/browse/CNX-8643